### PR TITLE
dmd.ctorflow: Remove self reference assignments in mergeFieldInit

### PR DESCRIPTION
**Pull request by ibuclaw**
@WalterBright - I'm not sure whether or not this was really a typo and it should have been `a = b`.

### CODE REVIEW

**Comment by WalterBright** [comment]
Yes, don't merge this until I figure it out.

**Comment by WalterBright** [comment]
Ok, I think I wrote it that way to emphasize what the result should be, and relied on the optimizer removing the self-assignment. So, the code is correct.

**Comment by ibuclaw** [comment]
> Ok, I think I wrote it that way to emphasize what the result should be, and relied on the optimizer removing the self-assignment. So, the code is correct.

I'll add a comment saying as much then, as the assignment alone is suspicious.

**Comment by WalterBright** [comment]
ok, good

**Comment by ibuclaw** [comment]
@WalterBright - I've added some notes throughout, let me know if something looks odd about them.

File: src/dmd/ctorflow.d [hunk 1]
```diff
@@ -190,29 +190,33 @@ bool mergeFieldInit(ref CSX a, const CSX b) pure nothrow
         return true;
     }
 
+    // The logic here is to prefer the branch that neither halts nor returns.
     bool ok;
     if (!bHalt && bRet)
     {
+        // Branch b returns, no merging required.
```
Comment (WalterBright) [inline src/dmd/ctorflow.d:197]: branch a returns

**Pull request merged**